src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -2538,3 +2593,64 @@ def policy_group_exists(self, group_name: str, zonegroup_name: str):
             return True
         except DashboardException:
             return False
+
+
+class RgwTopics:
+
+    def list_topics(self, uid: Optional[str], tenant: Optional[str]):
+        rgw_topics_list = {}
+        rgw_topic_list_cmd = ['topic', 'list']
+        try:
+            if uid:
+                rgw_topic_list_cmd.append('--uid')
+                rgw_topic_list_cmd.append(uid)
+
+            if tenant:
+                rgw_topic_list_cmd.append('--tenant')
+                rgw_topic_list_cmd.append(tenant)
+
+            exit_code, out, _ = mgr.send_rgwadmin_command(rgw_topic_list_cmd)
+            if exit_code > 0:
+                raise DashboardException(msg='Unable to fetch topic list',
+                                         http_status_code=500, component='rgw')
+            rgw_topics_list = out
```

Comment:
return as soon as finished.
```suggestion
            return rgw_topics_list
```

Reply:
"Suggested changes are done."